lib/src/phy/channel/channel.cc

```diff
@@ -94,6 +100,25 @@ channel::channel(const channel::args_t& channel_args, uint32_t _nof_channels, sr
     srsran_channel_rlf_init(rlf, channel_args.rlf_t_on_ms, channel_args.rlf_t_off_ms);
   }
 
+  // Create Tuner
+  if (channel_args.tuner_enable && ret == SRSRAN_SUCCESS) {
+    tuner = std::make_unique<srsran_channel_tuner_t>(logger);
+    // this->tuner->sock.open(channel_args.tuner_name);
+    tuner->tuner_monitor_thread = std::make_unique<std::thread>([this, &logger]{
```

Comment:
If you want tuner_monitor_thread to reside in "tuner", I suggest you to create it in the constructor of tuner class instead of here. Otherwise you may face memory problem on destruction.